docs/book/advanced-guide/pipelines/materializers.md

```diff
@@ -386,35 +375,21 @@ returned and consumed from pipeline steps, you might sometimes need to have a
 completely non-materialized artifact in a step, e.g., if you need to know the
 exact path to where your artifact is stored.
 
-A non-materialized artifact is a `BaseArtifact` (or any of its subclasses) and
-has a property `uri` that points to the unique path in the artifact store where
-the artifact is stored. One can use a non-materialized artifact by 
-specifying it as the type in the step:
+A non-materialized artifact is a `zenml.model.ArtifactResponseModel`. 
+Among others, it has a property `uri` that points to the unique path in the 
+artifact store where the artifact is persisted. One can use a non-materialized 
+artifact by specifying `ArtifactResponseModel` as the type in the step:
 
 ```python
-from zenml.artifacts import DataArtifact
+from zenml.models.artifact_models import ArtifactResponseModel
 from zenml.steps import step
 
 
 @step
-def my_step(my_artifact: DataArtifact):  # rather than pd.DataFrame
+def my_step(my_artifact: ArtifactResponseModel):  # rather than pd.DataFrame
```

Comment:
I don't like this too much, why would I as the user care (or even know) that this is somehow related to a response from something? I think this needs some rethinking and maybe some kind of public class that hides this information

Reply:
It also looked a bit out of place for me as well.

Reply:
Good point, will change this to use a new class `UnmaterializedArtifact` that subclasses `ArtifactResponseModel`.